src/dayNightSystem/dayNightClock.mjs

```diff
@@ -6,69 +6,101 @@
 // note: don't run more than one day/night clock at a time.  doing so will cause multiple
 //       filters to be applied to the screen, which won't look too nice. :o)
 
-import { Prim, Thread } from 'sphere-runtime';
+import { Thread } from 'sphere-runtime';
 
 import InGameTime from './inGameTime';
 
-const DayMask = Color.Transparent,
-      TwilightMask = new Color(0.5, 0.125, 0.0625, 0.625),
-      NightMask = new Color(0.0, 0.0, 0.125, 0.5625);
+const DayMask = [0.0, 0.0, 0.0, 0.0],
+      TwilightMask = [0.5, 0.125, 0.0625, 0.625],
+      NightMask = [0.0, 0.0, 0.125, 0.5625];
+
+const Day = 0,
+      Night = 1,
+      Fading = 2;
 
 export default
 class DayNightClock extends Thread
 {
 	constructor()
 	{
 		super({ priority: 1 });
+		this.inGameTime = new InGameTime(0, 0, 0);
 
 		console.log("initializing day/night clock", `time: ${this.now()}`);
-		this.currentMask = Color.Transparent;
+		let white = Color.White;
+		let width = Surface.Screen.width;
+		let height = Surface.Screen.height;
+		let rectangle = new Shape(ShapeType.TriStrip,
+			new VertexList([
+				{ x: 0, y: 0, color: white },
+				{ x: width, y: 0, color: white },
+				{ x: 0, y: height, color: white },
+				{ x: width, y: height, color: white },
+			]));
+		this.model = new Model ([rectangle]);
+		this.shader = new Shader({
+			fragmentFile: '#/shaders/image.frag.glsl',
+			vertexFile  : '#/shaders/image.vert.glsl',
+		});
+		this.model.shader = this.shader;
+		this.state = Day;
+		this.currentMask = [0.0, 0.0, 0.0, 0.0];
 		this.start();
 	}
 
 	now()
 	{
-		let realTime = new Date();
-		let currentTime = 3600 * realTime.getHours() + 60 * realTime.getMinutes() + realTime.getSeconds();
-		currentTime = (currentTime * 10) % 86400;
-		let hour = Math.floor(currentTime / 3600);
-		let minute = Math.floor((currentTime / 60) % 60);
-		let second = currentTime % 60;
-		return new InGameTime(hour, minute, second);
+		const time = Date.now();
+		let currentTime = Math.floor(time / 100) % 86400;
+		this.inGameTime.hour = Math.floor(currentTime / 3600);
+		this.inGameTime.minute = Math.floor((currentTime / 60) % 60);
+		this.inGameTime.second = currentTime % 60;
+		return this.inGameTime;
 	}
 
 	on_render()
 	{
-		Prim.fill(Surface.Screen, this.currentMask);
+		this.model.draw();
+	}
+
+	mixMasks(mask1, mask2, proportion1, proportion2)
+	{
+		this.currentMask[0] = mask1[0] * proportion1 + mask2[0] * proportion2;
+		this.currentMask[1] = mask1[1] * proportion1 + mask2[1] * proportion2;
+		this.currentMask[2] = mask1[2] * proportion1 + mask2[2] * proportion2;
+		this.currentMask[3] = mask1[3] * proportion1 + mask2[3] * proportion2;
```

Comment:
I wonder if we could do this in a shader somehow...

Reply:
I can't think of an efficient way to put this in a shader - any method I can think of would involve multiple uniforms and be  more costly overal.

It could perhaps be re-factored as some kind of more widely available helper function or method though rather than being just here.